aicsimageio/writers/ome_zarr_writer.py

```diff
@@ -0,0 +1,406 @@
+                            f"{len(channel_names)}"
+                        )
+            if channel_colors is not None:
+                if isinstance(channel_colors[0], list):
+                    if not isinstance(channel_colors[0][0], int):
+                        if len(channel_colors) != num_images:
+                            raise exceptions.ConflictingArgumentsError(
+                                f"OmeZarrWriter received a list of arrays to use as "
+                                f"scenes but the provided list of channel_colors is of "
+                                f"different length. "
+                                f"Number of provided scenes: {num_images}, "
+                                f"Number of provided dimension strings: "
+                                f"{len(channel_colors)}"
+                            )
+
+        # make sure data is a list
+        if not isinstance(data, list):
+            data = [data]
+        num_images = len(data)
+
+        # If metadata is attached as singles, expand to lists to match data
+        if dim_order is None or isinstance(dim_order, str):
+            dim_order = [dim_order] * num_images
+        if image_name is None or isinstance(image_name, str):
+            image_name = [image_name] * num_images
+        if isinstance(physical_pixel_sizes, tuple):
+            physical_pixel_sizes = [physical_pixel_sizes] * num_images
+        elif physical_pixel_sizes is None:
+            physical_pixel_sizes = [
+                types.PhysicalPixelSizes(None, None, None)
+            ] * num_images
+        if channel_names is None or isinstance(channel_names[0], str):
+            channel_names = [channel_names] * num_images  # type: ignore
+
+        if channel_colors is not None:
+            if all(
+                [
+                    (
+                        channel_colors[img_idx] is None
+                        or isinstance(channel_colors[img_idx], list)
+                    )
+                    for img_idx in range(num_images)
+                ]
+            ):
+                single_image_channel_colors_provided = False
+            else:
+                single_image_channel_colors_provided = True
+
+            if (
+                channel_colors[0] is not None
+                and isinstance(channel_colors[0], list)
+                and isinstance(channel_colors[0][0], int)
+            ):
+                single_image_channel_colors_provided = True
+
+        if channel_colors is None or single_image_channel_colors_provided:
+            channel_colors = [channel_colors] * num_images  # type: ignore
+
+        # Save image to zarr store!
+        mypath = pathlib.Path(uri)
+        # TODO handle overwrite scenario?
+        # shutil.rmtree(mypath)
+        # print(mypath)
+        store = parse_url(mypath, mode="w").store
+        # print(store)
+        root = zarr.group(store=store)
+        # print(root)
```

Comment:
Currently, attempts to overwrite raise exceptions deeper inside the zarr stack.  I haven't made this a priority to attempt to even diagnose yet.  I think it might be complicated because it won't rmdir the whole zarr ahead of time, and as a result you can be left with stray chunks all over the place.